waiter/src/waiter/state/responder.clj

```diff
@@ -482,9 +483,10 @@
   [{:keys [instance-id->state] :as current-state} update-instance-id->blacklist-expiry-time-fn update-status-tag-fn
    instance-id expiry-time]
   (log/info "unblacklisting instance" instance-id "as blacklist expired at" expiry-time)
-  (cond-> (update-instance-id->blacklist-expiry-time-fn current-state #(dissoc % instance-id))
```

Comment:
`cond->` returns the first argument unmodified if all the conditions are falsy. The `when` version will return `nil` instead.